src/dotnet/commands/dotnet-build/Program.cs

```diff
@@ -34,13 +35,67 @@ public static int Run(string[] args)
             }
         }
 
-        private static bool OnExecute(List<ProjectContext> contexts, CompilerCommandApp args)
+        private static bool OnExecute(IEnumerable<string> files, IEnumerable<NuGetFramework> frameworks, CompilerCommandApp args)
+        {
+            var builderCommandApp = (BuilderCommandApp)args;
+            var graphCollector = new ProjectGraphCollector(
+                !builderCommandApp.ShouldSkipDependencies,
+                (project, target) => ProjectContext.Create(project, target));
+
+            var contexts = ResolveRootContexts(files, frameworks, args);
+            var graph = graphCollector.Collect(contexts).ToArray();
+            var builder = new DotNetProjectBuilder(builderCommandApp);
+            return builder.Build(graph).ToArray().All(r => r != CompilationResult.Failure);
+        }
+
+        private static IEnumerable<ProjectContext> ResolveRootContexts(
+            IEnumerable<string> files,
+            IEnumerable<NuGetFramework> frameworks,
+            CompilerCommandApp args)
         {
-            var compileContexts = contexts.Select(context => new CompileContext(context, (BuilderCommandApp)args)).ToList();
 
-            var incrementalSafe = compileContexts.All(c => c.IsSafeForIncrementalCompilation);
+            // Set defaults based on the environment
+            var settings = ProjectReaderSettings.ReadFromEnvironment();
 
-            return compileContexts.All(c => c.Compile(incrementalSafe));
+            if (!string.IsNullOrEmpty(args.VersionSuffixValue))
+            {
+                settings.VersionSuffix = args.VersionSuffixValue;
+            }
+
+            foreach (var file in files)
+            {
+                var project = ProjectReader.GetProject(file);
+                var allFrameworks = project.GetTargetFrameworks().Select(f => f.FrameworkName);
+                if (!allFrameworks.Any())
+                {
+                    throw new InvalidOperationException(
+                        $"Project '{file}' does not have any frameworks listed in the 'frameworks' section.");
+                }
+                IEnumerable<NuGetFramework> selectedFrameworks;
+                if (frameworks != null)
+                {
+                    selectedFrameworks = allFrameworks.Where(f => frameworks.Any(ff => ff.Equals(f)));
```

Comment:
We do not support having multiple frameworks as `dotnet build` argument, its either null or one. I just made this class to support a little more just in case.